src/azure-cli/azure/cli/command_modules/vm/_image_builder.py

```diff
@@ -519,6 +526,100 @@ def create_image_template(  # pylint: disable=too-many-locals, too-many-branches
+                                                        user_assigned_identities=user_assigned_identities)
+        image_template_update = ImageTemplateUpdateParameters()
+        image_template_update.identity = image_template_identity
+        return patch_image_template(cmd.cli_ctx, resource_group_name, image_template_name, image_template_update)
+
+    image_template = assign_identity_helper(cmd.cli_ctx, getter, setter)
+    return image_template.identity
+
+
+def remove_template_identity(cmd, client, resource_group_name, image_template_name, user_assigned=None):
+    from azure.mgmt.imagebuilder.models import ImageTemplateUpdateParameters
+
+    def getter():
+        return client.virtual_machine_image_templates.get(resource_group_name, image_template_name)
+
+    def setter(resource_group_name, image_template_name, image_template):
+        image_template_update = ImageTemplateUpdateParameters(identity=image_template.identity)
+        return client.virtual_machine_image_templates.begin_update(resource_group_name, image_template_name,
+                                                                   image_template_update)
+
+    return _remove_template_identity(cmd, resource_group_name, image_template_name, user_assigned, getter, setter)
+
+
+def _remove_template_identity(cmd, resource_group_name, image_template_name, user_assigned, getter, setter):
+    resource = getter()
+    if resource is None or resource.identity is None:
+        return None
+
+    user_identities_to_remove = []
+    if user_assigned is not None:
+        existing_user_identities = {x.lower() for x in list((resource.identity.user_assigned_identities or {}).keys())}
+        # all user assigned identities will be removed if the length of user_assigned is 0,
+        # otherwise the specified identity
+        subscription_id = get_subscription_id(cmd.cli_ctx)
+        user_identities_to_remove = existing_user_identities if len(user_assigned) == 0 else set()
+        for ide in user_assigned:
+            if not is_valid_resource_id(ide):
+                ide = resource_id(subscription=subscription_id, resource_group=resource_group_name,
+                                  namespace='Microsoft.ManagedIdentity', type='userAssignedIdentities', name=ide)
+            user_identities_to_remove.add(ide.lower())
+
+        non_existing = user_identities_to_remove.difference(existing_user_identities)
+        if non_existing:
+            from azure.cli.core.azclierror import InvalidArgumentValueError
+            raise InvalidArgumentValueError("'{}' are not associated with '{}', please provide existing user managed "
+                                            "identities".format(','.join(non_existing), image_template_name))
+
+        if not list(existing_user_identities - user_identities_to_remove):
+            resource.identity.type = "None"
+
+    resource.identity.user_assigned_identities = None
```

Comment:
Do we need to put `resource.identity.user_assigned_identities = None` into the `if` statement?
```suggestion
        if not list(existing_user_identities - user_identities_to_remove):
            resource.identity.type = "None"
            resource.identity.user_assigned_identities = None
```